addons/barfiller/barfiller.lua

```diff
@@ -0,0 +1,88 @@
+--[[    BSD License Disclaimer
+        Copyright © 2015, Morath86
+        All rights reserved.
+
+        Redistribution and use in source and binary forms, with or without
+        modification, are permitted provided that the following conditions are met:
+
+            * Redistributions of source code must retain the above copyright
+              notice, this list of conditions and the following disclaimer.
+            * Redistributions in binary form must reproduce the above copyright
+              notice, this list of conditions and the following disclaimer in the
+              documentation and/or other materials provided with the distribution.
+            * Neither the name of BarFiller nor the
+              names of its contributors may be used to endorse or promote products
+              derived from this software without specific prior written permission.
+
+        THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+        ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+        WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+        DISCLAIMED. IN NO EVENT SHALL Morath86 BE LIABLE FOR ANY
+        DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+        (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+        LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+        ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+        (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+        SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'BarFiller'
+_addon.author = 'Morath'
+_addon.version = '0.1.7'
+_addon.commands = {'bf','barfiller'}
+_addon.language = 'english'
+
+--Windower Libs
+config = require('config')
+file = require('files')
+packets = require('packets')
+math = require('math')
+
+--BarFiller Libs
+require('statics')
+
+-- Generate Settings Files
+-- Thanks to Byrth & SnickySnacks' BattleMod addon
+settings_file = file.new('data\\settings.xml',true)
+settings_table = config.load('data\\settings.xml',DefaultSettings)
+config.save(settings_table)
+
+initialize()
+
+-- Addon commands
+-- Thanks to Byrth & SnickySnacks' BattleMod addon
+windower.register_event('addon command',function(...)
+    local commands = {...}
+    local first_cmd = table.remove(commands,1):lower()
```

Comment:
Instead of doing this, there's a simpler way:
```lua
windower.register_event('addon command', function(command, ...)
    command = command and command:lower() or 'help'
    local args = {...}

    -- Rest here
end)
```

This will automatically make the first value the command and set it to `'help'` if it doesn't exist (your approach would currently result in a Lua error if the first command was missing).